stdlib/websub/src/main/ballerina/src/websub/hub_service.bal

```diff
@@ -504,6 +508,26 @@ returns error? {
     return;
 }
 
+# Function to retrieve cached subscriberCallbackClient for a given callback.
+#
+# + callback - The callback URL registered for the subscriber
+# + return - `http:Client` indicating the client for a given callback from cache or new client
+function getSubcriberCallbackClient(string callback) returns http:Client {
+    http:Client subscriberCallbackClient;
+    if (subscriberCallbackClientCache.hasKey(callback)) {
+        return <http:Client>subscriberCallbackClientCache.get(callback);
+    } else {
+        lock {
```

Comment:
why is this part locked and the if block not locked?

Reply:
This block creates the http:Client and lock is to avoid multiple threads creating multiple clients for the same callback.
If block does not need the lock, as it's just a read operation.